examples/retry_command.ts

```diff
@@ -0,0 +1,34 @@
+import { Command, CommandStep, Pipeline } from '../src';
+import { YamlSerializer } from '../src/serializers/yaml';
+
+class RetryCommand extends Command {
+    constructor(public retries: number, public c: Command) {
+        super(c.toString());
+        this.retries = retries;
+    }
+
+    protected serialize(): string {
+        return new Array(this.retries + 1).fill(this.c).join(' || ');
+    }
+
+    public toString(): string {
+        return this.serialize();
+    }
+}
+
+const install = new RetryCommand(1, new Command('yarn'));
+
+const lint = new CommandStep([
+    install,
+    new RetryCommand(1, new Command('yarn lint')),
+]).withKey('lint');
+
+const test = new CommandStep([install, new Command('yarn test', 2)])
+    .withKey('test')
+    .dependsOn(lint);
+
+const pipeline = new Pipeline('My pipeline').add(lint).add(test);
```

Comment:
```suggestion

const pipeline = new Pipeline('My pipeline').add(step);
```

I don't think adding another command adds anything to the example.